packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -170,6 +170,11 @@ export class TableRow<
         return null;
     }
 
+    public override disconnectedCallback(): void {
```

Comment:
Do we need to consider the lifecycle where a row element is disconnected and then reconnected? If yes, what would cause the columns to be re-observed in that case? Would `columnsChanged()` be called again?

Reply:
Good point, didn't think through it but if discussion about thinking of the full connect / disconnect lifecycle is important that should be captured in the common PR items memory leaks comment: https://github.com/ni/nimble/issues/1479#issuecomment-2344585509

Reply:
I don't think we need to support that, given that `nimble-table-row` is just an implementation detail of the `nimble-table` (not exposed via the table's public API) and not otherwise intended for use by clients.

`columnsChanged()` does not get called upon re-connecting a row to the DOM. There could be other state in the `TableRow` (e.g. `columns`) that went stale while the component was disconnected, so supporting disconnect/reconnect would take more than simply adding a call to `observeColumns()` in `connectedCallback`.